libraries/helpers.rb

```diff
@@ -307,5 +307,25 @@ def package_options
 
       options
     end
+
+    #
+    # Checks the deprecated properties of chef-ingredient and prints warning
+    # messages if any of them are being used.
+    #
+    def check_deprecated_properties
```

Comment:
So the basic rule here is: _Dropping the leading `chef-*` or `opscode-*` unless the product name is `chef-server` or `chef-backend`_. We can express that as:

```ruby
def check_deprecated_properties
  if !(%w(chef-server chef-backend).include?(new_resource.product_name)) && 
      (match = new_resource.product_name.match(/(chef-|opscode-)(?<product_key>.*)/))

    new_product_key = match[:product_key]
    Chef::Log.warn "product_name '#{new_resource.product_name}' is deprecated and it will be removed in the future versions of chef-ingredient. Use '#{new_product_key}' instead of '#{new_resource.product_name}'."
    new_resource.product_name(new_product_key)
  end
end
```

We can actually greatly simplify this method (and save future additions to the `deprecated_product_names ` hash).